<launch>
    
    <!--- Sim Time -->
    <!-- <param name="/use_sim_time" default="false" /> -->
    <arg name="input_initial_pose_topic" default="/initialpose" doc="Initial position topic to align" />
    <arg name="odom_topic" default="/odometry/filtered/global" doc="robot odom topic" />
    <arg name="submap_size_xy" default="50.0" />
    <arg name="submap_size_z" default="20.0" />
    <arg name="map_switch_thres" default="25.0" />

    <arg name="roll" default="0.0" />
    <arg name="pitch" default="0.0" />
    <arg name="yaw" default="0.0" />
    <arg name="x" default="0.0" />
    <arg name="y" default="0.0" />
    <arg name="z" default="0.0" />


    <!-- <arg name="roll" default="1.570795" />
    <arg name="pitch" default="0.0" />
    <arg name="yaw" default="1.570795" />
    <arg name="x" default="0.0" />
    <arg name="y" default="0.0" />
    <arg name="z" default="0.0" /> -->

    <!--- Run Rviz-->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find ndt_localizer)/cfgs/jaguar.rviz" />

    <!--- MapLoader -->    
    <arg name="pcd_path"  default="$(find ndt_localizer)/map/court_yard_map.pcd"/>
    <!-- <arg name="pcd_path"  default="/media/rdcas/dataset/map_result/kaist02.pcd"/> -->
    
    <arg name="map_topic" default="/points_map"/>


    <node pkg="ndt_localizer" type="map_loader"    name="map_loader"    output="screen">
        <param name="pcd_path" value="$(arg pcd_path)"/>
        <param name="map_topic" value="$(arg map_topic)"/>
        <param name="robot_pose_topic" value="$(arg odom_topic)"/>
        <param name="init_pose_topic" value="$(arg input_initial_pose_topic)"/>

        <param name="submap_size_xy" value="$(arg submap_size_xy)"/>
        <param name="submap_size_z" value="$(arg submap_size_z)"/>
        <param name="map_switch_thres" value="$(arg map_switch_thres)"/>

        <param name="roll" value="$(arg roll)" />
        <param name="pitch" value="$(arg pitch)" />
        <param name="yaw" value="$(arg yaw)" />
        <param name="x" value="$(arg x)" />
        <param name="y" value="$(arg y)" />
        <param name="z" value="$(arg z)" />
    </node>

</launch>
